apps/web/pages/api/twilio/webhook.ts

```diff
@@ -0,0 +1,140 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import { CreditService } from "@calcom/features/ee/billing/credit-service";
+import * as twilio from "@calcom/features/ee/workflows/lib/reminders/providers/twilioProvider";
+import { IS_SMS_CREDITS_ENABLED, WEBAPP_URL } from "@calcom/lib/constants";
+import getOrgIdFromMemberOrTeamId from "@calcom/lib/getOrgIdFromMemberOrTeamId";
+import { defaultHandler } from "@calcom/lib/server";
+import prisma from "@calcom/prisma";
+
+/*
+  Twilio status callback: creates expense log when sms is delivered or undelivered
+*/
+async function handler(req: NextApiRequest, res: NextApiResponse) {
+  const signature = req.headers["x-twilio-signature"];
+  const baseUrl = `${WEBAPP_URL}/api/twilio/webhook`;
+
+  const queryParams = new URLSearchParams(req.query as Record<string, string>).toString();
+  const requestUrl = queryParams ? `${baseUrl}?${queryParams}` : baseUrl;
+
+  if (typeof signature === "string") {
+    const isSignatureValid = await twilio.validateWebhookRequest({
+      requestUrl,
+      signature,
+      params: req.body,
+    });
+    if (isSignatureValid) {
```

Comment:
`nit:` can we early return error if signature is not a string or invalid?

Reply:
Good call! Added some early returns to improve readability